build/Dockerfile

```diff
@@ -100,12 +100,10 @@ WORKDIR /tmp
 RUN python -m pip install --upgrade pip
 
 # TODO Move to installing wheel once we have proper releases setup instead of cloning the repo
-RUN git clone https://github.com/foundation-model-stack/fms-hf-tuning.git && \
-    cd fms-hf-tuning && \
-    python -m pip install ".[dev]" && \
-    python -m pip install ".[flash-attn]" && \
-    python -m pip install ".[aim]" && \
-    python -m pip install -U datasets
+ARG WHEEL
+COPY ${WHEEL} ${WHEEL}
+RUN python -m pip install "${WHEEL}[dev,aim]" && \
+    python -m pip install "${WHEEL}[flash-attn]"
```

Comment:
I think ideally the dev dependencies would not be installed into the image each time. I see we were already doing this but ideally this would be a multistage dockerfile with a build stage that includes the dev dependencies for building the wheel. 

Also, what is `WHEEL` supposed to be set to? Within the image, we should be building the wheel and running `python -m build --wheel --outdir bdist`

Reply:
If I only install [flash-attn] directly, I got No `packaging` module error.  The packaging module needs to be installed by the [dev] first.
With this Dockerfile change, the way we build image will become:
```
python -m build --wheel .   # creates dist/fms_hf_tuning-0.0.1-py3-none-any.whl
docker build -t sft-trainer:dev --build-arg WHEEL=dist/fms_hf_tuning-0.0.1-py3-none-any.whl . -f build/Dockerfile
```

Reply:
@anhuong I revised the Dockerfile so the build command is same as before for example: `docker build -t fms-hf-tuning:dev . -f build/Dockerfile`